Automattic/jetpack
# [RNMobile] Implement useSyncMedia hook on native

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5611

## Proposed changes:

* The `useSyncMedia` hook has been called from [the `edit.native.js` file](https://github.com/Automattic/jetpack/pull/29996/files#diff-80669e45a350ffdb93b81e6ca442d6faaa68f0d2d7e34298862893d61f73ff50), so that the block's settings are synced across web and the app.
* https://github.com/Automattic/jetpack/pull/29996/commits/6b2e83630e1ed4c8c56527ececfb9af52cdbae89: A temporary "Save Settings" button has been introduced. **Note, this is temporary and will not be shipped to users. We aim to remove this hack from the code within a week.** The purpose of this button is to enable us to more easily test the sync functionality internally.

### Other information:


## Jetpack product discussion

p9ugOq-3mV-p2 

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* With the changes from this PR applied, first navigate to the post editor on the web.
* Either add a new VideoPress block or use an existing one, and make sure to tweak its settings. It's particularly important to tweak the settings in the **Details** and **Privacy and rating** panels, as these are the settings that are not currently being synced.
* Save your changes on the web and then navigate to the post in the native app.
* Verify that the block's settings have been correctly fetched from the web.
* Next, update the block's settings in the app itself. 
* Tap the temporary **Save Settings** button from the settings panel. 
* Verify that these changes have been synced to the web.
* Go through a few tries of editing the block on the web then returning the app, and vice versa, to ensure syncing is working as expected.

**Screen recording:**

https://user-images.githubusercontent.com/2998162/231724361-a4850ce7-8413-4b29-943a-d74804feab32.mov
